### ISSUE TITLE
Is it possible to support referencing a published toml file as the source?

### ISSUE THREAD

**Issue by qinkangdeid** [austinarbor/version-catalog-generator#107]
Is it possible to support referencing a published toml file as the source? For example, publishing it to an internal Nexus. I see that this TomlConfig can only specify a file.
```kotlin
  class TomlConfig(private val settings: Settings) {
        /** The name of the library in the TOML catalog file */
        lateinit var libraryAlias: String

        /** The catalog file containing the BOM library entry */
        var file: File =
            settings.rootDir.toPath().resolve(Paths.get("gradle", "libs.versions.toml")).toFile()

        internal fun isInitialized(): Boolean {
            return ::libraryAlias.isInitialized
        }
    }

```

**Comment by austinarbor** [comment]
Hi @qinkangdeid , if you have a TOML published to maven / nexus, you actually don't need this plugin at all, you can use built-in Gradle behavior. For example, Micronaut publishes their catalog to maven and it can be consumed / used like this

```kotlin
dependencyResolutionManagement {
    repositories {
        mavenCentral()
    }
    versionCatalogs {
        create("mn") {
            from("io.micronaut:micronaut-bom:${micronautVersion}")
        }
    }
}
```

The way it works is they have the toml file published under the BOM coordinates [micronaut-bom-3.10.3](https://repo1.maven.org/maven2/io/micronaut/micronaut-bom/3.10.3/) and the standard Gradle version catalog plugin knows how to find it

**Comment by qinkangdeid** [comment]
My original intention was: I have a published toml file that defines the springboot dependencies library like this:

```toml
[versions]
springBootVersion = "some version"
[libraries]
spring-boot-dependencies = { group = "org.springframework.boot", name = "spring-boot-dependencies", version.ref = "springBootVersion" }

```

I want to publish it and then use your plugin to reference this library. I just hope to be able to use the version number variable`springBootVersion` defined in my published toml file without having to redefine it in the project using the plugin.

However, maybe I can also combine my original toml file with this plugin and then publish a merged custom plugin for projects to use.

**Comment by qinkangdeid** [comment]
> My original intention was: I have a published toml file that defines the springboot dependencies library like this:
> 
> ```toml
> [versions]
> springBootVersion = "some version"
> [libraries]
> spring-boot-dependencies = { group = "org.springframework.boot", name = "spring-boot-dependencies", version.ref = "springBootVersion" }
> ```
> 
> I want to publish it and then use your plugin to reference this library. I just hope to be able to use the version number variable`springBootVersion` defined in my published toml file without having to redefine it in the project using the plugin.
> 
> However, maybe I can also combine my original toml file with this plugin and then publish a merged custom plugin for projects to use.

It should look like this reference:
```kotlin
generate("springLibs") {
      from {
        dependency("org.springframework.boot:spring-boot-dependencies:3.1.2") 
      }
    }
```

But for the `3.1.2` part, I want to reference what's defined in the published toml file.

**Comment by austinarbor** [comment]
I think Fetching the whole TOML from a remote URL would probably be easier than just the variable value for the version. What kind of authentication would you need to download from the Nexus URL?

**Comment by qinkangdeid** [comment]
Yes, I think so too. Based on the previous usage, enabling the file to support fetching files from the network and reading the Nexus file in the generate method should be feasible.
Currently, my Nexus configuration looks like this(setting.gradle.kts): 
```groovy
dependencyResolutionManagement {
    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
    repositories {
        mavenLocal()
        maven {
            // config in user gradle dir  gradle.properties (group repositore)  https://example.com/repository/public/     
            url = uri(providers.gradleProperty("nexusRepositoryUrl").get())
            credentials {
                username = providers.gradleProperty("nexusUsername").get()
                password = providers.gradleProperty("nexusPassword").get()
            }
        }
        mavenCentral()
        gradlePluginPortal()
    }
    versionCatalogs {
        // this is the published catalog
        create("libs") {
            from("com.example:catalog:1.0.0")
        }
    }
}

```

like `from("com.example:catalog:1.0.0")` dependencyNotation config?

```groovy

generate("springLibs") {
    from {
        toml {
            libraryAlias = "spring-boot-dependencies" 
            file = file("com.example:catalog:1.0.0")
            // or:  file = remoteFile("com.example:catalog:1.0.0") 
        }
    }
}

```

**Comment by austinarbor** [comment]
Yup, I think I should be able to get that to work. I can probably repurpose some of the logic [here](https://github.com/austinarbor/version-catalog-generator/blob/main/plugin/src/main/kotlin/dev/aga/gradle/versioncatalogs/service/GradleDependencyResolver.kt#L48-L60) and change `pom` to `toml` to get it to resolve your toml file from Nexus, and I'll add some sort of `remoteFile` option like you mentioned. I'm on vacation until Monday but will take a shot at this next week

**Comment by austinarbor** [comment]
@qinkangdeid i have released 1.5.0 which now has this feature, let me know how it works for you!
```kotlin
generate("springLibs") {
  from {
    toml {
      libraryAlias = "spring-boot-dependencies" 
      file = artifact("com.example:catalog:1.0.0")
    }
  }
}
```

**Comment by qinkangdeid** [comment]
Yes, it worked after the upgrade. Fantastic!
